### ISSUE TITLE
[bug] fold will result in wrong position

### ISSUE THREAD

**Issue by Freed-Wu** [psliwka/vim-smoothie#34]
```vim
" $ uname -r
" 5.16.15-arch1-1
" $ has nvim
" ✔ nvim 0.6.1
" $ has python
" ✔ python 3.10.2
" $ cat test.vim
set runtimepath=$VIMRUNTIME
set runtimepath+=~/.local/share/nvim/repos/github.com/psliwka/vim-smoothie
nmap gg    <Plug>(Smoothie_gg)
" $ vi -u test.vim ~/.config/nvim/init.vim
```

![Screenshot from 2022-04-02 19-44-19](https://user-images.githubusercontent.com/32936898/161381545-efcb4837-18c1-4e54-b79f-55b34572208f.png)

Press `33gl`, the position is still line 1.

![Screenshot from 2022-04-02 19-44-31](https://user-images.githubusercontent.com/32936898/161381621-9b890bc2-4399-4b55-95cb-6040c86dc244.png)

However, if you comment the last line of test.vim. the position is correct line 33.

**Comment by psliwka** [comment]
Hi @Freed-Wu! I've pushed a potential fix in f2aaab110b49da21ab7abf4858173e3194109f6a. Could you please re-test it in your setup?

**Comment by Freed-Wu** [comment]
I can confirm the bug has been fixed.
